modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+// Module included in the following assemblies:
+//
+// * web_console/dynamic-plug-ins-reference.adoc
+
+:_content-type: CONCEPT
+[id="dynamic-plug-in-api_{context}"]
+= {product-title} console API
+
+[discrete]
+== `useActivePerspective`
+
+Hook that provides the currently active perspective and a callback for setting the active perspective. It returns a tuple containing the current active perspective and setter callback.
+
+[source,tsx]
+----
+const Component: React.FC = (props) => {
+   const [activePerspective, setActivePerspective] = useActivePerspective();
+   return <select
+     value={activePerspective}
+     onChange={(e) => setActivePerspective(e.target.value)}
+   >
+     {
+       // ...perspective options
+     }
+   </select>
+}
+----
+
+[discrete]
+== `GreenCheckCircleIcon`
+
+Component for displaying a green check mark circle icon.
+
+[source,tsx]
+----
+<GreenCheckCircleIcon title="Healthy" />
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`className` |(optional) additional class name for the component
+|`title` |(optional) icon title
+|`size` |(optional) icon size: ('sm', 'md', 'lg', 'xl')
+|===
+
+[discrete]
+== `RedExclamationCircleIcon`
+
+Component for displaying a red exclamation mark circle icon.
+
+[source,tsx]
+----
+<RedExclamationCircleIcon title="Failed" />
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`className` |(optional) additional class name for the component
+|`title` |(optional) icon title
+|`size` |(optional) icon size: ('sm', 'md', 'lg', 'xl')
+|===
+
+[discrete]
+== `YellowExclamationTriangleIcon`
+
+Component for displaying a yellow triangle exclamation icon.
+
+[source,tsx]
+----
+<YellowExclamationTriangleIcon title="Warning" />
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`className` |(optional) additional class name for the component
+|`title` |(optional) icon title
+|`size` |(optional) icon size: ('sm', 'md', 'lg', 'xl')
+|===
+
+[discrete]
+== `BlueInfoCircleIcon`
+
+Component for displaying a blue info circle icon.
+
+[source,tsx]
+----
+<BlueInfoCircleIcon title="Info" />
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`className` |(optional) additional class name for the component
+|`title` |(optional) icon title
+|`size` |(optional) icon size: ('sm', 'md', 'lg', 'xl')
+|===
+
+[discrete]
+== `ErrorStatus`
+
+Component for displaying an error status popover.
+
+[source,tsx]
+----
+<ErrorStatus title={errorMsg} />
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`title` |(optional) status text
+|`iconOnly` |(optional) if true, only displays icon
+|`noTooltip` |(optional) if true, tooltip won't be displayed
```

Comment:
s/won't/is not

Reply:
@mburke5678 Everything from the dynamic-plug-in-api.adoc file comes from API doc comments in the openshift/console repo (for example, https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts). We'll need to update the API doc there and refresh here. Can we make this a follow on?

Reply:
@spadgett As far as I am concerned. @opayne1 and I chatted on Slack earlier about this very matter.